state-synchronizer/src/request_manager.rs

```diff
@@ -403,7 +409,8 @@ impl RequestManager {
         }
 
         // increment multicast level if this request is also multicast-timed-out
-        if Self::is_timeout(last_request_time, self.multicast_timeout) {
+        let multicast_start_time = self.get_multicast_start_time(version).unwrap_or(UNIX_EPOCH);
```

Comment:
multicast_start_time is always <= last_request_time, so this would trigger the timeout more often?

Reply:
multicast timeout and regular timeout are tracked differently, regular timeout is computed earlier in the code, and they each have different timeout durations. This is just checking for multicast timeout and updating state accordingly